capstone/src/main/webapp/youtubeScript.js

```diff
@@ -0,0 +1,176 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+const ATTRIBUTES_BY_LANGUAGE = {
+  'en': ['TOXICITY', 'SEVERE_TOXICITY', 'TOXICITY_FAST', 'IDENTITY_ATTACK', 'INSULT', 'PROFANITY', 'THREAT', 'SEXUALLY_EXPLICIT', 'FLIRTATION'],
+  'es': ['TOXICITY', 'SEVERE_TOXICITY', 'IDENTITY_ATTACK_EXPERIMENTAL', 'INSULT_EXPERIMENTAL', 'PROFANITY_EXPERIMENTAL', 'THREAT_EXPERIMENTAL'],
+  'fr': ['TOXICITY', 'SEVERE_TOXICITY', 'IDENTITY_ATTACK_EXPERIMENTAL', 'INSULT_EXPERIMENTAL', 'PROFANITY_EXPERIMENTAL', 'THREAT_EXPERIMENTAL'],
+  'de': ['TOXICITY', 'SEVERE_TOXICITY', 'IDENTITY_ATTACK', 'INSULT', 'PROFANITY', 'THREAT'],
+  'it': ['TOXICITY', 'SEVERE_TOXICITY', 'IDENTITY_ATTACK', 'INSULT', 'PROFANITY', 'THREAT'],
+  'pt': ['TOXICITY', 'SEVERE_TOXICITY', 'IDENTITY_ATTACK', 'INSULT', 'PROFANITY', 'THREAT']
+};
+
+/** Calls youtube servlet and passes output to perspctive */
+async function callYoutube() {
+  const channelId = document.getElementById('channelIdForAnalysis').value.replace(/ /g, '');
+  if (!channelId) {
+    return;
+  }
+  var response;
+  /** Checks if input follows channel ID format, if not attempts to convert it to channel ID*/
+  if (channelId[0] == "U" && channelId[1] == "C" && channelId.length == 24 && isLetter(channelId[channelId.length-1])) {
+    response = await fetch('/youtube_servlet?channelId=' + channelId,)
+    response = await response.json();
+    if (response.hasOwnProperty('error')) {
+      alert("Invalid Channel ID");
+      inputCommentsToPerspective([]);
+      return;
+    }
+  } else {
+    const usernameConverterResponse = await fetch('/username_servlet?channelId=' + channelId,)
+    const usernameConverterResponseJson = await usernameConverterResponse.json();
+    if (usernameConverterResponseJson.pageInfo.totalResults == 0) {
+      alert("Username Not found, Please Input Channel ID");
+      inputCommentsToPerspective([]);
+      return;
+    }
+    const convertedUserName = usernameConverterResponseJson.items[0].id;
+    response = await fetch('/youtube_servlet?channelId=' + convertedUserName,)
+    response = await response.json();
+  }
+  inputCommentsToPerspective(response);
+}
+
+/** Calls perspective to analyze comments */
+async function inputCommentsToPerspective(comments) {
+  const langElement = document.getElementById('languageForAnalysis');
+  if (!langElement) {
+    return;
+  }
+  const commentListElement = document.getElementById('comment-list');
+  commentListElement.innerHTML = '';
+  const requestedAttributes = getRequestedAttributes();
+  if (!requestedAttributes) {
+      return;
+  }
+  const attributeScores = [];
+  for (const item in comments.items) {
+    const perspectiveScore = await callPerspective(comments.items[item].snippet.topLevelComment.snippet.textOriginal, langElement.value, requestedAttributes);
+    attributeScores.push(perspectiveScore);
+  }
+  const attributeTotals = new Map();
```

Comment:
Nit: this `inputCommentsToPerspective` function is a little long, you might consider splitting it into 2 or more separate pieces.  One option might be to create a separate function like `getAttributeAverages(attributeScores)` which takes the responses from `callPerspective` and creates the Map that you'll need to pass to `loadChartsApi`

Reply:
addressed in "address host and intern Daniel's nits"